File: mario.c
// Draws a pyramid using iteration

#include <cs50.h>
#include <stdio.h>

int get_positive_int(void);

int main(void)
{
    // Get height of pyramid form user
    int h = get_positive_int();

    // Draw pyramid using user's input
    for (int i = 1; i <= h; i++)
    {
        for (int k = i; k < h; k++)
        {
            printf(" ");
        }
        for (int j = 1; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

// Prompt user for positive integer within one and eight
int get_positive_int(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    return n;
}

File: mario.py
# get input from user
h = int(input("Height: "))

# Ensure that users typed in a positive integer between 1 and 8, inclusive.
while h < 1 or h > 8:
    print("provide a positive integer between 1 and 8, inclusive.")
    h = int(input("Height: "))

# Draw a pyramid of height h
for i in range(h):
    for j in range(i+1, h, 1):
        print(" ", end="")
    for k in range(i+1):
        print("#", end="")
    print(sep="")
